python/ray/tests/test_autoscaler_gcp.py

```diff
@@ -0,0 +1,62 @@
+from typing import List
+
+import pytest
+
+from ray.autoscaler._private.gcp.node_provider import _retry
+
+
+class MockGCPNodeProvider:
+    def __init__(self, errors: List[type]):
+        # List off errors to raise while retrying self.mock_method
+        self.errors = errors
+        # Incremented during each retry via self._construct_client
+        self.error_index = -1
+        # Mirrors the __init__ of GCPNodeProvider
+        # Also called during each retry in _retry
+        self._construct_clients()
+
+    def _construct_clients(self):
+        # In real life, called during each retry to reinitializes api clients.
+        # Here, increments index in list of errors passed into test.
+        self.error_index += 1
+
+    @_retry
+    def mock_method(self, *args, **kwargs):
+        error = self.errors[self.error_index]
+        if error:
+            raise error
+        return (args, kwargs)
+
+
+def test_gcp_broken_pipe_retry():
+    # Short names for two types of errors
+    b, v = BrokenPipeError, ValueError
+    # BrokenPipeError is supposed to caught with up to 5 tries.
+    # ValueError is an arbitrarily chosen exception which should not be caught.
+
+    error_inputs = [
+        [None],  # No error, success
+        [b, b, b, b, None],  # Four failures followed by success
+        [b, b, v, b, None],  # ValueError raised
+        [b, b, b, b, b, None],  # max 5 tries allowed,raise
+        [b, b, b, b, b, b, None],  # also raise
+    ]
+    expected_errors_raised = [None, None, v, b, b]
```

Comment:
Can you use a fixture here with tuples? That will make debugging/the output easier to parse.